From def0f2513e7550ef7d37c82f15f1d265d1c9e73c Mon Sep 17 00:00:00 2001
From: kumarama-i <aman.kumar@intel.com>
Date: Tue, 25 Jan 2022 09:26:10 +0530
Subject: [PATCH 1/2] tty: 8250_lpss: check for NULL function ptr

Check for setup() func availability before calling it, similar handling with exit()

Signed-off-by: Tan, Raymond <raymond.tan@intel.com>
Signed-off-by: Shevchenko, Andriy <andriy.shevchenko@linux.intel.com>
---
 drivers/tty/serial/8250/8250_lpss.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/tty/serial/8250/8250_lpss.c b/drivers/tty/serial/8250/8250_lpss.c
index d3bafec7619d..1d0a67bebd4b 100644
--- a/drivers/tty/serial/8250/8250_lpss.c
+++ b/drivers/tty/serial/8250/8250_lpss.c
@@ -324,9 +324,11 @@ static int lpss8250_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (!uart.port.membase)
 		return -ENOMEM;
 
-	ret = lpss->board->setup(lpss, &uart.port);
-	if (ret)
-		return ret;
+	if (lpss->board->setup) {
+		ret = lpss->board->setup(lpss, &uart.port);
+		if (ret)
+			return ret;
+	}
 
 	dw8250_setup_port(&uart.port);
 
-- 
2.30.2

